packages/web-runtime/src/components/Topbar/UserMenu.vue

```diff
@@ -93,7 +117,86 @@ export default defineComponent({
     }
   },
   setup() {
+    const checkboxValue = ref(false);
+    const notificationsModalOpen = ref(false)
+    const store = useStore()
+    const accessToken = useAccessToken({ store })
+
+    const updateNotificationPreference = (option) => {
+      const headers = new Headers()
+      headers.append('Authorization', `Bearer ${unref(accessToken)}`)
+      headers.append('X-Requested-With', 'XMLHttpRequest')
+
+      fetch('/ocs/v1.php/cloud/user', {
+        method: 'PATCH',
+        headers,
+        body: JSON.stringify({ disableNotifications: option })
+      })
+        .then((response) => {
+          if (response.ok) {
+            store.commit('SET_NOTIFICATION', option)
+            closeNotificationsModal()
+          } else {
+            throw new Error('Notification setting could not be applied')
+          }
+        })
+        .catch((err) => {
+          store.dispatch('showMessage', {
+            title: 'An error occurred',
+            desc: err || 'Notification setting could not be applied',
+            status: 'danger'
+          })
+        })
+      return
+    }
+
+    const getNotificationPreference = () => {
```

Comment:
You can rewrite this function using `async/await` syntax which will make it clearer, you can take a look [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) for more info.